lib/http2/hpack.c

```diff
@@ -410,14 +485,18 @@ int h2o_hpack_parse_headers(h2o_req_t *req, h2o_hpack_header_table_t *header_tab
                             const char **err_desc)
 {
     const uint8_t *src_end = src + len;
+    int parsing_ret = 0;
 
     *content_length = SIZE_MAX;
 
     while (src != src_end) {
         struct st_h2o_decode_header_result_t r;
         int ret = decode_header(&req->pool, &r, header_table, &src, src_end, err_desc);
```

Comment:
I think you would need to create a local value for `err_desc` initialized to NULL, and pass pointer to the created value when calling `decode_header`, so that you could determine the cause of correctly memoize the error status in `decode_header`.

In current implementation it seems that you might mis-record errors for headers following an invalid header.